pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -187,9 +193,15 @@ private synchronized SegmentCompletionFSM lookupOrCreateFsm(final LLCSegmentName
     final String instanceId = reqParams.getInstanceId();
     final long offset = reqParams.getOffset();
     LLCSegmentName segmentName = new LLCSegmentName(segmentNameStr);
-    SegmentCompletionFSM fsm = lookupOrCreateFsm(segmentName, SegmentCompletionProtocol.MSG_TYPE_COMMIT);
-    SegmentCompletionProtocol.Response response = fsm.segmentCommitStart(instanceId, offset);
-    if (fsm.isDone()) {
+    SegmentCompletionFSM fsm = null;
```

Comment:
they are calling different methods into the fsm, so i could not find a way to share it.